### PR TITLE
Ensure restore map is a valid git repo

### PR DESCRIPTION
**Pull request by yufeih**
- Do not checkout original branch when bilingual is on (only fetch)
- Use `HEAD` file to check if a worktree is valid
- `GetGitRemoteInfo` -> `SplitGitHref` to align with `SplitHref`

### CODE REVIEW

File: src/docfx/restore/RestoreGit.cs [hunk 1]
```diff
@@ -54,9 +54,16 @@ async Task AddWorkTrees()
 
                     await ParallelUtility.ForEach(branches, async branch =>
                     {
+                        // Bilingual repos ({branch}-sxs) only depend on non bilingual branch for git commit history,
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreGit.cs]: that makes the convention everywhere...

File: src/docfx/restore/RestoreMap.cs [hunk 2]
```diff
@@ -37,18 +37,24 @@ public string GetGitRestorePath(string url)
...
-                    .OrderByDescending(f => new DirectoryInfo(f).LastWriteTimeUtc)
-                    .FirstOrDefault();
+                return (
+                    from path in Directory.EnumerateDirectories(restoreDir, "*", SearchOption.TopDirectoryOnly)
+                    let name = Path.GetFileName(path)
+                    where name.StartsWith(HrefUtility.EscapeUrlSegment(branch) + "-") &&
+
+                        // Adding the HEAD file is the last step in a worktree checkout, use it to
+                        // filter out worktrees that are still in progress.
+                        File.Exists(Path.Combine(restoreDir, ".git/worktrees", name, "HEAD"))
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreMap.cs]: do we have a doc to reference?

File: src/docfx/restore/RestoreGit.cs (hunk 1 above)
Comment (live1206) [reply src/docfx/restore/RestoreGit.cs]: how about having a method in `LocalizationConvention.cs` to hide `-sxs`:
```csharp
public static string GetBilingualBranch(string branch) => $"{branch}-sxs";
```
Anywhere we need to access it, just use this one.

File: src/docfx/restore/RestoreMap.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/restore/RestoreMap.cs]: there is no doc about it, I digged into list worktree and add worktree implementation

Comment (live1206) [reply src/docfx/restore/RestoreMap.cs]: Maybe we can comment the related code link to our code for reference

**Pull request merged**